<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>FormScriptGenerator</class>
 <widget class="QWidget" name="FormScriptGenerator">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1348</width>
    <height>731</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Script Generator</string>
  </property>
  <property name="windowIcon">
   <iconset resource="qBorgPilot.qrc">
    <normaloff>:/images/borgpilot_16x16.png</normaloff>:/images/borgpilot_16x16.png</iconset>
  </property>
  <layout class="QGridLayout" name="gridLayout_3">
   <item row="0" column="0">
    <widget class="QFrame" name="frame_ToolBar">
     <property name="frameShape">
      <enum>QFrame::Shape::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Shadow::Raised</enum>
     </property>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QTabWidget" name="tabWidget_ScriptGen">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab_Init">
      <attribute name="title">
       <string>Tab 1</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_10">
       <item row="0" column="0" colspan="2">
        <widget class="QGroupBox" name="groupBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>250</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="title">
          <string>Script Name</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_8">
          <item row="1" column="4">
           <widget class="QToolButton" name="toolButton_ScrGen">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Creates a new script&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>...</string>
            </property>
            <property name="icon">
             <iconset theme="QIcon::ThemeIcon::MediaPlaybackStart"/>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QToolButton" name="toolButton_ScrOpen">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Load definitions&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>...</string>
            </property>
            <property name="icon">
             <iconset theme="document-open"/>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLineEdit" name="lineEdit_ScriptName">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Max length: 15 bytes&lt;/p&gt;&lt;p&gt;Valid chars: a-z 0-9 _&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="maxLength">
             <number>15</number>
            </property>
            <property name="clearButtonEnabled">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="3">
           <widget class="QToolButton" name="toolButton_ScrSave">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Save definitions&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>...</string>
            </property>
            <property name="icon">
             <iconset theme="document-save"/>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QGroupBox" name="groupBox_RepoNames">
         <property name="title">
          <string>Repositories Name</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_7">
          <item row="0" column="0" rowspan="2">
           <widget class="QTableWidget" name="tableWidget_RepoNames">
            <property name="alternatingRowColors">
             <bool>true</bool>
            </property>
            <property name="selectionMode">
             <enum>QAbstractItemView::SelectionMode::SingleSelection</enum>
            </property>
            <property name="selectionBehavior">
             <enum>QAbstractItemView::SelectionBehavior::SelectRows</enum>
            </property>
            <column>
             <property name="text">
              <string>Name</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>::Archive</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Encryption mode</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Append only</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Make path dirs</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Storage quota (GiB)</string>
             </property>
            </column>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>Common Options (Applies to all repositories)</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_9">
          <item row="0" column="0">
           <widget class="TreeView_CommonOptions_P" name="treeView_CommonOptions_Init">
            <property name="alternatingRowColors">
             <bool>true</bool>
            </property>
            <property name="wordWrap">
             <bool>false</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QFrame" name="frame_2">
         <property name="frameShape">
          <enum>QFrame::Shape::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Shadow::Raised</enum>
         </property>
         <layout class="QGridLayout" name="gridLayout_6">
          <item row="0" column="0">
           <layout class="QGridLayout" name="gridLayout_5">
            <item row="0" column="0">
             <widget class="QRadioButton" name="radioButton_Local">
              <property name="text">
               <string>Local</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QRadioButton" name="radioButton_Remote">
              <property name="toolTip">
               <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;A key must be configured for ssh to access the remote server.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
              <property name="text">
               <string>Remote</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="0" column="1">
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>498</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="1" column="0" colspan="2">
           <widget class="QLineEdit" name="lineEdit_DestPath">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Max. length: 254 bytes&lt;/p&gt;&lt;p&gt;Valid chars: Aa-Zz 0-9 _ /&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="maxLength">
             <number>254</number>
            </property>
            <property name="readOnly">
             <bool>false</bool>
            </property>
            <property name="placeholderText">
             <string>Destination path</string>
            </property>
            <property name="clearButtonEnabled">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_Create">
      <attribute name="title">
       <string>Tab 2</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <widget class="QGroupBox" name="groupBox_3">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>350</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="title">
          <string>Script Name</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_11">
          <item row="1" column="3">
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="1" column="7">
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="1" column="0">
           <widget class="QLineEdit" name="lineEdit_ScriptCreateName">
            <property name="minimumSize">
             <size>
              <width>136</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>136</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Max length: 15 bytes&lt;/p&gt;&lt;p&gt;Valid chars: a-z 0-9 _&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="maxLength">
             <number>15</number>
            </property>
            <property name="clearButtonEnabled">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QToolButton" name="toolButton_CreateOpen">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Load initialialization script definitions&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>...</string>
            </property>
            <property name="icon">
             <iconset theme="document-open"/>
            </property>
           </widget>
          </item>
          <item row="1" column="8">
           <widget class="QToolButton" name="toolButton_CreateGen">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Generate a new script&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>...</string>
            </property>
            <property name="icon">
             <iconset theme="QIcon::ThemeIcon::MediaPlaybackStart"/>
            </property>
           </widget>
          </item>
          <item row="1" column="5">
           <widget class="QToolButton" name="toolButton_CreateReload">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Reload definitions&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>...</string>
            </property>
            <property name="icon">
             <iconset theme="QIcon::ThemeIcon::EditUndo"/>
            </property>
           </widget>
          </item>
          <item row="1" column="6">
           <widget class="QToolButton" name="toolButton_CreateSave">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Save definitions&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>...</string>
            </property>
            <property name="icon">
             <iconset theme="document-save"/>
            </property>
           </widget>
          </item>
          <item row="1" column="10">
           <widget class="QToolButton" name="toolButton_ShowLog">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Show log file&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>...</string>
            </property>
            <property name="icon">
             <iconset theme="QIcon::ThemeIcon::DocumentPrintPreview"/>
            </property>
           </widget>
          </item>
          <item row="1" column="9">
           <spacer name="horizontalSpacer_4">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="treeWidget_CreateBackup_P" name="treeWidget_CreateBackup">
         <property name="alternatingRowColors">
          <bool>true</bool>
         </property>
         <column>
          <property name="text">
           <string notr="true">Repositories</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Source Path</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Common Options</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Optional Options</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Data</string>
          </property>
         </column>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>TreeView_CommonOptions_P</class>
   <extends>QTreeView</extends>
   <header location="global">treeview_commonoptions_p.h</header>
  </customwidget>
  <customwidget>
   <class>treeWidget_CreateBackup_P</class>
   <extends>QTreeWidget</extends>
   <header location="global">treewidget_createbackup_p.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>tabWidget_ScriptGen</tabstop>
  <tabstop>lineEdit_ScriptName</tabstop>
  <tabstop>toolButton_ScrOpen</tabstop>
  <tabstop>toolButton_ScrGen</tabstop>
  <tabstop>radioButton_Local</tabstop>
  <tabstop>radioButton_Remote</tabstop>
  <tabstop>lineEdit_DestPath</tabstop>
  <tabstop>tableWidget_RepoNames</tabstop>
  <tabstop>treeView_CommonOptions_Init</tabstop>
 </tabstops>
 <resources>
  <include location="qBorgPilot.qrc"/>
 </resources>
 <connections/>
</ui>
